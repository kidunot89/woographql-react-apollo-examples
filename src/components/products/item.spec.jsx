import { render } from '@testing-library/react';

import Item from './item';

describe('Item component', () => {
  const mockData = {
    uri: '/product/product',
    name: 'Product',
    onSale: true,
    regularPrice: '$20.00',
    price: '$18.99',
    image: {
      sourceUrl: 'https://place-puppy.com/300x300',
      altText: 'It\'s a dog',
    },
    galleryImages: [],
    type: 'SIMPLE',
    shortDescription: 'short description'
  }
  const { baseElement } = render(<Item data={mockData} />);

  it('renders successfully', () => {
    expect(baseElement).toMatchSnapshot();
  });
});
