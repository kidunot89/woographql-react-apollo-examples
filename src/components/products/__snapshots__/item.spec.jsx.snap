// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component renders successfully 1`] = `
<body>
  <div>
    <a
      class="sc-bczRLJ ihqCpe"
      direction="column"
      href="/product/product"
      type="normal"
    >
      <img
        alt="It's a dog"
        class="sc-dkzDqf ehJGnl"
        src="https://place-puppy.com/300x300"
        width="175px"
      />
      <div
        alignitems="center"
        class="sc-bczRLJ sc-gsnTZi kEqOsw daxFCl"
        direction="column"
        height="175px"
        width="175px"
      >
        <div
          class="product-name"
        >
          <span
            class="badge"
          >
            On Sale
          </span>
          <br />
          Product
          <br />
          short description
          <p
            class="product-price"
          >
            <span
              class="regular-price"
            >
              $20.00
            </span>
             
            $18.99
          </p>
        </div>
      </div>
    </a>
  </div>
</body>
`;
